shuttle/src/runtime/execution.rs

```diff
@@ -28,6 +28,57 @@ scoped_thread_local! {
     static EXECUTION_STATE: RefCell<ExecutionState>
 }
 
+// The reason this is separated out from `ExecutionState` is to ensure that we're always able to persist the schedule.
+// If we don't do this, then we may panic while borrowing `ExecutionState`, and then not be able to emit the schedule.
+// If we then panic again while trying to handle the panic, such that the panic becomes an abort, we will never log
+// the schedule.
+//
+// It is expected that if the `ExecutionState` exists, then this will exist, and any usage of this happens through the
+// `ExecutionState`, or at a point where it is known that the `ExecutionState` must exist (eg. when serializing on a panic).
+thread_local! {
+    static CURRENT_SCHEDULE: CurrentSchedule = CurrentSchedule::default();
+}
+
+#[derive(Debug, Default)]
+pub struct CurrentSchedule {
+    // SAFETY: There are two ways having an `UnsafeCell` could be unsafe:
+    //         1: Multiple threads are accessing the `UnsafeCell` at the same time, and at least one thread does a write.
+    //         2: We give out a reference or a pointer to the `UnsafeCell`, and that reference is used after the contents have been moved or freed.
+    //         These cannot occur, as Shuttle is single threaded, and we never give out a reference or a raw pointer.
+    current_schedule: UnsafeCell<Schedule>,
```

Comment:
I'm not convinced we need to use `UnsafeCell` here.  Why is `RefCell` not enough?

Reply:
It may be okay today, but someone might change something tomorrow and break the SAFETY assumptions.  In general, I'd prefer to avoid having to think about this stuff unless absolutely necessary.